### ISSUE TITLE
Error to parse docker-compose v3 format

### ISSUE THREAD

**Issue by pivulic** [kubernetes-incubator/kompose#412]
## Problem
When trying to create named volumes that can be reused across multiple services (without relying on volumes_from due to "[PersistentVolumeClaim is not bound](http://stackoverflow.com/questions/41738374/kubernestes-persistentvolumeclaim-is-not-bound)"), kompose is unable to parse docker-compose.yml version 3 format. 

See: https://docs.docker.com/compose/compose-file/#/volume-configuration-reference

## How To Recreate
1. Create file:`touch docker-compose.yml`
1. Copy/paste:

    ```    
    version: "3"
    services:
      db:
        image: db
        volumes:
          - data-volume:/var/lib/db
      backup:
        image: backup-service
        volumes:
          - data-volume:/var/lib/backup/data

      volumes:
        data-volume:
    ```
1. Run: `kompose up -f docker-compose.yml`

## Actual Result
```
ERRO Could not parse config for project test : yaml: unmarshal errors:
  line 1: cannot unmarshal !!str `3` into config.RawService
FATA Failed to load compose file: yaml: unmarshal errors:
  line 1: cannot unmarshal !!str `3` into config.RawService
```
## Expected Result
docker-compose.yml should successfully be converted and deployed.

**Comment by procrypt** [comment]
@pivulic `Kompose` does not support docker-compose version 3 yet.

**Comment by michaelharrer** [comment]
when will it be available? (roundabout)

**Comment by cdrage** [comment]
@surajssd Hey man, do you know if this will be updated upstream in libcompose?

**Comment by cdrage** [comment]
Ahhhh, just found the issue: https://github.com/docker/libcompose/issues/421

We'll have to wait for this to be updated upstream @michaelharrer @pivulic

**Comment by surajssd** [comment]
yeah v3 is still WIP in upstream and I guess once that is done we can move to it.

**Comment by cdrage** [comment]
@michaelharrer @pivulic 

The best way to "workaround" this is simply removing the "deploy" section of your V3 file and changing the `v3` value to `v2` and it'll work fine. Luckily there aren't any *too* major changes between the versions :)

**Comment by pivulic** [comment]
What "deploy" section? :) The reason why I'm trying the `v2` version, is to be able to use [named volumes](https://docs.docker.com/compose/compose-file/#/volume-configuration-reference), instead of using [volumes_from](https://docs.docker.com/compose/compose-file/#/volumesfrom), trying to get around the "[PersistentVolumeClaim is not bound](http://stackoverflow.com/questions/41738374/kubernestes-persistentvolumeclaim-is-not-bound)") issue.
But I'm going to try and create the Persistent Volume in my cluster before running `kompose up` (using a `v2` docker-compose.yml), as you recommend in https://github.com/kubernetes-incubator/kompose/issues/411#issuecomment-277693584 :)

**Comment by cdrage** [comment]
@pivulic Ahhhhh. I got my facts wrong. I thought the only large difference was the `deploy` section of v3. They added named volumes too! We'll see what we can do to fix it. Funny enough, another issue was just opened in regards to using underscores and such in volume names (bug), we could possibly fix two issues with one PR :)

**Comment by cdrage** [comment]
I'm labeling this as a bug with libcompose as this will need to be supported upstream as well as downstream within Kompose.

**Comment by surajnarwade** [comment]
@surajssd @kadel , can you please add `compose v3` label here ?

**Comment by cdrage** [comment]
@surajnarwade done

**Comment by cdrage** [comment]
@pivulic #600 has now been merged! :tada: :tada: :tada:

**Comment by pivulic** [comment]
Wohoo!
